src/main/scala/com/box/castle/core/leader/TaskManager.scala

```diff
@@ -1,22 +1,29 @@
 package com.box.castle.core.leader
 
-import com.box.castle.core.config.{LeaderConfig, CommitterConfig}
+import com.box.castle.committer.api.TopicFilter
+import com.box.castle.core.config.{CommitterConfig, LeaderConfig}
 import com.box.castle.core.worker.tasks.Task
 import org.slf4s.Logging
 
-import scala.collection.immutable.{TreeMap, Seq}
+import scala.collection.immutable.{Seq, TreeMap}
 import scala.util.Random
 import scala.util.matching.Regex
 import scala.collection.mutable
 
 object TaskManager {
-  def apply(leaderConfig: LeaderConfig, committerConfigs: Iterable[CommitterConfig]) =
-    new TaskManager(leaderConfig, committerConfigs)
+  def apply(
+             leaderConfig: LeaderConfig,
+             committerConfigs: Iterable[CommitterConfig],
+             topicFilterMap: Map[String, TopicFilter]) =
```

Comment:
Every commiterFactory has a TopicFilter associated with it. It should be mandatory